### PR TITLE
[Zen2] Migrate no-master-block integration tests

### PR DESCRIPTION
**Pull request by DaveCTurner**
This change follows up on #36478 by migrating the affected integration tests to
use Zen2.

### CODE REVIEW

**Review by andrershov — COMMENTED** [review]
Looks good to me, except Zen setting, which ideally should not be used.

File: server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java [hunk 1]
```diff
@@ -53,104 +58,113 @@
...
     public void testNoMasterActions() throws Exception {
         Settings settings = Settings.builder()
-                .put("action.auto_create_index", true)
-                .put("discovery.zen.minimum_master_nodes", 2)
-                .put(ZenDiscovery.PING_TIMEOUT_SETTING.getKey(), "200ms")
-                .put("discovery.initial_state_timeout", "500ms")
-                .put(DiscoverySettings.NO_MASTER_BLOCK_SETTING.getKey(), "all")
-                .build();
+            .put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(), true)
+            .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), Integer.MAX_VALUE)
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java]: Unfortunately today `InternalTestCluster` currently requires that it's explicitly set if `autoMinMasterNodes` is `false`, which is why it's present but set ludicrously high. Once we've finished migrating everything to Zen2 it will be easy enough to find all usages like this to remove them.

**Comment by DaveCTurner** [comment]
@elasticmachine please run the Gradle build tests 2

File: server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java [hunk 2]
```diff
@@ -53,104 +58,113 @@
...
+            .put(GatewayService.RECOVER_AFTER_MASTER_NODES_SETTING.getKey(), 3)
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java]: why add this setting? This should not be needed?

File: server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java [hunk 3]
```diff
@@ -179,19 +193,19 @@ void checkWriteAction(ActionRequestBuilder<?, ?> builder) {
...
         Settings settings = Settings.builder()
-                .put("action.auto_create_index", false)
-                .put("discovery.zen.minimum_master_nodes", 2)
-                .put(ZenDiscovery.PING_TIMEOUT_SETTING.getKey(), "200ms")
-                .put("discovery.initial_state_timeout", "500ms")
-                .put(DiscoverySettings.NO_MASTER_BLOCK_SETTING.getKey(), "write")
-                .build();
+            .put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(), false)
+            .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), Integer.MAX_VALUE)
+            .put(GatewayService.RECOVER_AFTER_MASTER_NODES_SETTING.getKey(), 3)
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java]: same here. Why is this needed?

File: server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java (hunk 2 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java]: It defaults to whatever `minimum_master_nodes` is. I could just set that to 3.

Comment (ywelsch) [reply server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java]: ah that's annoying, and does not make any sense in Zen2. Can you perhaps change GatewayService to not do this if `discovery instanceof Coordinator`?

Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java]: Ok, I pushed b0e9643e686b262f65697bbd2d8f111cfc6e18e9.

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Apparently-unrelated failures.

@elasticmachine please run the gradle build tests 1
